### PR TITLE
Context manager for constants

### LINKED ISSUE #6950: Context manager for constant versions
For some use cases it would be helpful to have a context manager to set the version set of the constants. E.g., something like 
```
with constants_set(astropyconst13):
    ... code goes here ...
````

### PR DESCRIPTION
**Pull request by pllim**
Fix #6950

### CODE REVIEW

File: astropy/constants/utils.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Utility functions for ``constants`` sub-package."""
```
Comment (pllim) [inline astropy/constants/utils.py]: @adrn , @eteq thinks you might appreciate this.

**Comment by mhvk** [comment]
Overall, this looks nice! One question, maybe not do-able at all, but would it be possible to do `with const.astropyconst13`, i.e., at some level have each constants sub-module be a context? (I hasten to add that I only partially understand contexts, but perhaps one can define `__enter__` and `__exit__` even on a module?)

**Comment by mhvk** [comment]
On the name, I ended up thinking we should just have a clear name, not use the modules. How about `set_enabled_constants`? This in analogy with `u.set_enabled_units`. Eventually, we might then also have `add_enabled_constants`.

Indeed, is there room for re-using some of the units constants machinery?

**Comment by pllim** [comment]
Ops. I made a mental note to add docstring and must have forgotten about it!

@mhvk -- That sounds like an elegant idea but won't make it to feature freeze. If you think it is worth pursuing instead of this one, perhaps we can re-milestone for 3.1 unless you would like to take a stab at it by Friday.

Thank you for all your comments!

**Comment by mhvk** [comment]
@pllim - how about just calling it `set_enabled_constants` - we can then extend for 3.1, but won't have to change the name. That version in units works the same as what you have: nothing is available except what you explicitly allow.

**Comment by pllim** [comment]
@stargaser and @mhvk , I think I have addressed the comments thus far. I agree that improvements could be made as a follow-up of this PR and has added a note to this effect in the code.

As for only supporting `astropyconst13` in the PR, I thought I would start small (the minimal amount of support that is sufficient) and see how things go. If there is enough demand to justify adding more, we can always expand in the future. I am afraid to support arbitrary combo now (CODATA2010 + IAU2015) would overly complicate things, and I am not even sure if anyone would use such a combo. So, that is my thought process for the record.

**Comment by stargaser** [comment]
I am satisfied with these changes, and the plan for following up in version 3.1.

**Comment by pllim** [comment]
@mhvk , do you have any objections? If not, please approve and merge. Thanks!

**Pull request merged**